api/client/src/test/java/org/projectnessie/client/TestNessieClientConfigSources.java

```diff
@@ -0,0 +1,206 @@
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Properties;
+import java.util.function.Function;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.junit.jupiter.InjectSoftAssertions;
+import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+
+@ExtendWith(SoftAssertionsExtension.class)
+public class TestNessieClientConfigSources {
+  @InjectSoftAssertions protected SoftAssertions soft;
+
+  @Test
+  void mapSource() {
+    soft.assertThat(asMap(mapConfigSource(emptyMap()), "a", "b"))
+        .containsEntry("a", null)
+        .containsEntry("b", null);
+    soft.assertThat(asMap(mapConfigSource(singletonMap("a", "foo")), "a", "b"))
+        .containsEntry("a", "foo")
+        .containsEntry("b", null);
+    soft.assertThat(
+            asMap(
+                mapConfigSource(ImmutableMap.of("a", "foo", "b", "bar", "c", "baz")),
+                "a",
+                "b",
+                "c"))
+        .containsEntry("a", "foo")
+        .containsEntry("b", "bar")
+        .containsEntry("c", "baz");
+  }
+
+  @Test
+  void propertiesSource() {
+    soft.assertThat(asMap(propertiesConfigSource(new Properties()), "a", "b"))
+        .containsEntry("a", null)
+        .containsEntry("b", null);
+
+    Properties p = new Properties();
+    p.put("a", "foo");
+    soft.assertThat(asMap(propertiesConfigSource(p), "a", "b"))
+        .containsEntry("a", "foo")
+        .containsEntry("b", null);
+
+    p = new Properties();
+    p.put("a", "foo");
+    p.put("b", "bar");
+    p.put("c", "baz");
+    soft.assertThat(asMap(propertiesConfigSource(p), "a", "b", "c"))
+        .containsEntry("a", "foo")
+        .containsEntry("b", "bar")
+        .containsEntry("c", "baz");
+  }
+
+  @Test
+  void environmentSource() {
+    soft.assertThat(asMap(environmentConfigSource(emptyMap()), "a.b", "b.c"))
+        .containsEntry("a.b", null)
+        .containsEntry("b.c", null);
+    soft.assertThat(asMap(environmentConfigSource(singletonMap("A_B", "foo")), "a.b", "b.c"))
+        .containsEntry("a.b", "foo")
+        .containsEntry("b.c", null);
+    soft.assertThat(
+            asMap(
+                environmentConfigSource(
+                    ImmutableMap.of("A_B", "foo", "B_C", "bar", "C_D_MINUS", "baz")),
+                "a.b",
+                "b.c",
+                "c.d-minus"))
+        .containsEntry("a.b", "foo")
+        .containsEntry("b.c", "bar")
+        .containsEntry("c.d-minus", "baz");
+  }
+
+  @Test
+  void environmentFileSource(@TempDir Path dir) throws IOException {
+    Path f1 = dir.resolve("c_1");
+    Path f2 = dir.resolve("c_2");
+    Path f3 = dir.resolve("c_3");
+    Files.createFile(f1);
+    Files.write(f2, singletonList("A_B=foo"));
+    Files.write(f3, asList("A_B=foo", "B_C=bar", "C_D_MINUS=baz"));
```

Comment:
Maybe have an integer value too?

Reply:
For strings?

Reply:
just to make sure they are parsed correctly

Reply:
I see, you do not trust `Properties.load()` ;)